Proposal: Fyne Tools Integration Images

lucor opened the pull request:
> ⚠️ **Warning** If accepted, these images will no longer be compatible with `fyne-cross`, which should now be considered **deprecated**. 
>
> Tag format changes:
>  - OLD: `docker.io/fyneio/fyne-cross-images:v1.0.0-android`
>  - NEW: `docker.io/fyneio/fyne-cross-images:tools-v1.0.0-android`
 

This PR proposes changes to support **Fyne tool integration**.


## Platform Support (Status)

- [x] **Linux:**  
- [x] **Windows:**  
- [ ] **Darwin (macOS):**  
- [ ] **FreeBSD:** 
- [ ] **Android:**  

## Changes

### Base Image Updates
- Upgrade base image from **Debian Bookworm** to **Debian Trixie**  
- Bump **Go** version from **1.25.3** to **1.25.5**  
- Remove **Fyne CLI** and **fyne-cross-s3** tool installations  
- Remove **fixuid** installation and Docker user setup

lucor committed:
feat(base): image update

This commit updates Go to v1.25.3, Zig to v0.15.2. Additionally move to
fyne tools v1.7.0

lucor committed:
feat(darwin-sdk-extractor): update to be compatible with zig 0.15.2

lucor committed:
fix(base): rollback to fyne 2.5.5

To use fyne tools an updated version of fyne-cross is needed.
Rollback to fyne CLI.

lucor committed:
feat(base): initial version for fyne tool integration

This commit updates the base docker image for the integration with fyne
tool:
- Update base image from Debian Bookworm to Debian Trixie
- Bump Go version from 1.25.3 to 1.25.5
- Remove Fyne CLI and fyne-cross-s3 tool installations from base image
- Remove fixuid installation and docker user setup

andydotxyz commented:
> ⚠️ Warning If accepted, these images will no longer be compatible with fyne-cross, which should now be considered deprecated.

I think this is a big deal - and if we use the same namespace then the deprecated tool will keep getting updates which will then break it.
Instead should we move to a new namespace or repo to avoid such breaks?

lucor committed:
feat: add PREFIX to image tags for fyne tool integration

Replace image tagging format to support PREFIX-based namespacing.

Tag format changes:
  - OLD: docker.io/fyneio/fyne-cross-images:v1.0.0-android
  - NEW: docker.io/fyneio/fyne-cross-images:tools-v1.0.0-android

PREFIX is configurable: PREFIX=custom make all

lucor commented:
> > ⚠️ Warning If accepted, these images will no longer be compatible with fyne-cross, which should now be considered deprecated.
> 
> I think this is a big deal - and if we use the same namespace then the deprecated tool will keep getting updates which will then break it. Instead should we move to a new namespace or repo to avoid such breaks?

Thanks for the catch @andydotxyz, forgot to update the namespace.

Updated to use a temporary namespace so we can test things safely without breaking the current `fyne-cross` setup:

`docker.io/fyneio/fyne-cross-images:tools-v1.0.0-android`

If the proposal is accepted, we can choose a proper new namespace (or repo) to avoid any conflict with the old `fyne-cross` images.

andydotxyz reviewed (commented):
I might be wrong here, but when building the legacy images the PREFIX would be "" but there is then a "-" leading to an extra character before version number on old image definitions.

Apologies if I am wrong about that - am no expert in Docker / Makefile
